frontends/p4/fromv1.0/converters.cpp

```diff
@@ -176,6 +176,53 @@ const IR::Node* ExpressionConverter::postorder(IR::GlobalRef *ref) {
             new IR::Path(ref->srcInfo, IR::ID(ref->srcInfo, ref->toString())));
 }
 
+/// P4_16 is stricter on comparing booleans with ints
+/// Therefore we convert such expressions into simply the boolean test
+const IR::Node* ExpressionConverter::postorder(IR::Equ *equ) {
+  const IR::Expression *boolExpr = nullptr;
+  const IR::Expression *constExpr = nullptr;
+  if (equ->left->type->is<IR::Type_Boolean>() &&
+      (equ->right->type->is<IR::Type_InfInt>() && equ->right->is<IR::Constant>())) {
```

Comment:
Why check for InfInt here?  Something like `hdr.valid == 1w1` would have a constant that is not an InfInt.  Something like `hdr.valid == true` is legal already but could be simplified to just `hdr.valid`.

It seems like just removing the check for InfInt would deal with all these cases.  Something like `hdr.valid == "foo"` should fail typechecking before this.